router.go

```diff
@@ -63,39 +72,76 @@ func (m MapRouter) Register(rs []transport.Procedure) {
 			panic("Expected procedure name not to be empty string in registration")
 		}
 
+		if r.Encoding != "" {
```

Comment:
why don't we just register everything in the encoding list?  That way we have a single source of truth for *all* procedures.